src/esp/assets/ResourceManager.cpp

```diff
@@ -2184,5 +2185,209 @@ std::unique_ptr<MeshData> ResourceManager::createJoinedCollisionMesh(
+    const std::string& MeshMetaDataFile,
+    const std::string& new_filename,
+    const std::string& filepath) {
+  bool success = Cr::Utility::Directory::mkpath(filepath);
+
+  const MeshMetaData& metaData = getMeshMetaData(MeshMetaDataFile);
+  Cr::Utility::Directory::writeString(filepath + "/" + new_filename,
+                                      "# CHD Mesh group \n");
+  // write vertex info to file
+  int numVertices = 0;
+  for (MeshTransformNode node : metaData.root.children) {
+    CollisionMeshData& meshData =
+        meshes_.at(node.meshIDLocal + metaData.meshIndex.first)
+            ->getCollisionMeshData();
+    for (auto& pos : meshData.positions) {
+      Cr::Utility::Directory::appendString(filepath + "/" + new_filename,
+                                           "v " + std::to_string(pos[0]) + ' ' +
+                                               std::to_string(pos[1]) + ' ' +
+                                               std::to_string(pos[2]) + '\n');
+      numVertices++;
+    }
+  }
+  Cr::Utility::Directory::appendString(
+      filepath + "/" + new_filename,
+      "# Number of vertices: " + std::to_string(numVertices) + "\n\n");
+
+  // Now do second pass to write indices for each group (node)
+  int globalVertexNum = 1;
+  int numParts = 1;
+  for (MeshTransformNode node : metaData.root.children) {
+    CollisionMeshData& meshData =
+        meshes_.at(node.meshIDLocal + metaData.meshIndex.first)
+            ->getCollisionMeshData();
+    Cr::Utility::Directory::appendString(
+        filepath + "/" + new_filename,
+        "g part_" + std::to_string(numParts) + " mesh\n");
+    for (int ix = 0; ix < meshData.indices.size(); ix += 3) {
+      Cr::Utility::Directory::appendString(
+          filepath + "/" + new_filename,
+          "f " + std::to_string(meshData.indices[ix] + globalVertexNum) + ' ' +
+              std::to_string(meshData.indices[ix + 1] + globalVertexNum) + ' ' +
+              std::to_string(meshData.indices[ix + 2] + globalVertexNum) +
+              '\n');
+    }
+    numParts++;
+    globalVertexNum += meshData.positions.size();
+  }
+
+  return success;
+}
+
+void ResourceManager::convexHullDecomposition(const std::string& filename,
+                                              const std::string& CHDFilename,
+                                              const VHACDParameters& params) {
+  if (resourceDict_.count(filename) == 0) {
+    // load/check_for render mesh metadata and load assets
+    bool renderMeshSuccess =
+        loadObjectMeshDataFromFile(filename, filename, "render", true);
+
+  }  // if no render asset exists
+
+  // fill points and triangles with mesh data
+  std::vector<float> points = std::vector<float>();
+  std::vector<uint32_t> triangles = std::vector<uint32_t>();
+  getPrimitiveMeshData(filename, points, triangles);
+
+  // use VHACD
+  bool res =
+      interfaceVHACD->Compute(&points[0], (unsigned int)(points.size() / 3),
```

Comment:
Instead of casting back and for shouldn't the divison just been done between two unsigned ints instead?

Reply:
Also, we should be trying to avoid C-Style cast's when possible right @aclegg3 ? Also there is a lot of unecessarily throwing away the signedness of ints in this PR. No reason to do narrowing conversions particularly here where you widen the conversion immediately after.